types/dnssrv.pp

```diff
@@ -0,0 +1,2 @@
+# match DNS RR (DNS SRV) similar to what is described by RFC2782 but is a superset to match real world usage by Freeipa. See https://www.rfc-editor.org/rfc/rfc2782
+type Stdlib::Dnssrv = Pattern[/\A(([_a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*(([_a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\z/]
```

Comment:
thinking about extensability i think it would be better to name space this as Stdlib::Dns::Srv